src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -239,7 +239,9 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
                 {
                     // Create the comment file for reshelve.
                     StringBuilder comment = new StringBuilder(tfShelveset.Comment ?? string.Empty);
-                    if (!(executionContext.Variables.Build_GatedRunCI ?? true))
+                    string runCi = GetEndpointData(endpoint, Constants.EndpointData.GatedRunCI);
+                    bool? gatedRunCi = StringUtil.ConvertToBoolean(runCi);
```

Comment:
ConvertToBoolean does not return `bool?`, it returns `bool`. You need to pass `defaultValue: true`, otherwise you're changing the logic here (previous code defaults to true).